extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/contract_agreements/ContractAgreementPageApiService.java

```diff
@@ -29,15 +31,25 @@ public class ContractAgreementPageApiService {
     private final ContractAgreementPageCardBuilder contractAgreementPageCardBuilder;
 
     @NotNull
-    public ContractAgreementPage contractAgreementPage() {
+    public ContractAgreementPage contractAgreementPage(@Nullable ContractAgreementPageQuery contractAgreementPageQuery) {
         var agreements = contractAgreementDataFetcher.getContractAgreements();
 
         var cards = agreements.stream()
-                .map(agreement -> contractAgreementPageCardBuilder.buildContractAgreementCard(
-                        agreement.agreement(), agreement.negotiation(), agreement.asset(), agreement.transfers()))
-                .sorted(Comparator.comparing(ContractAgreementCard::getContractSigningDate).reversed())
-                .toList();
+            .map(agreement -> contractAgreementPageCardBuilder.buildContractAgreementCard(
+                agreement.agreement(),
+                agreement.negotiation(),
+                agreement.asset(),
+                agreement.transfers(),
+                agreement.terminations()))
```

Comment:
question: can you have multiple terminations per contract agreement?

Reply:
Or is this a map of all contract agreement IDs to their respective statuses? But why wouldn't you just fetch the termination status together with the rest of the metadata by joining the tables?

Or am I reading the code wrong?